lib/password_manager_2.rb

```diff
@@ -1,2 +1,46 @@
+require 'date'
+
 class PasswordManager2
+    def initialize
+        @all_entries = []
+    end
+
+    def add(new_service, new_password)
+        if @all_entries.any?{|entry| entry["service"] == new_service}
+            return "ERROR: Service names must be unique"
+        elsif @all_entries.any?{|entry| entry["password"] == new_password}
+            return "ERROR: Passwords must be unique"
+        else
+            new_entry = {"service" => new_service, "password" => new_password, "added_on" => DateTime.now(), "updated" => DateTime.now()}
+            @all_entries.push(new_entry)
+        end
+    end
+
+    def remove(service_name)
+        entry_to_be_removed = @all_entries.find{|entry| entry["service"] == service_name}
```

Comment:
**Praise:** Very good and easily understandable naming of the variables - makes it easier to understand when working with the same codebase later on.

Reply:
Noted! Thanks :)